core/chaincode/lifecycle/metadata_provider.go

```diff
@@ -129,6 +131,30 @@ func (mp *MetadataProvider) Metadata(channel string, ccName string, collections
 		CollectionsConfig:  ccInfo.Definition.Collections,
 	}
 
+	for _, col := range collections {
+		if isImplicit, mspID := implicitcollection.MspIDIfImplicitCollection(col); isImplicit {
+			// Implicit collection
+
+			// Initialize definition of collections if it's missing
+			if ccInfo.Definition.Collections == nil {
+				ccInfo.Definition.Collections = &peer.CollectionConfigPackage{}
+			}
+
+			ccInfo.Definition.Collections.Config = append(ccInfo.Definition.Collections.Config, &peer.CollectionConfig{
+				Payload: &peer.CollectionConfig_StaticCollectionConfig{
+					StaticCollectionConfig: &peer.StaticCollectionConfig{
+						Name: col,
+						MemberOrgsPolicy: &peer.CollectionPolicyConfig{
+							Payload: &peer.CollectionPolicyConfig_SignaturePolicy{
+								SignaturePolicy: policydsl.SignedByMspMember(mspID),
+							},
+						},
```

Comment:
Shouldn't you add here `EndorsementPolicy` field as well which would carry the same value as for the `MemberOrgsPolicy`. Look at the following for loop that populates the map `ccMetadata.CollectionPolicie` with `EndorsementPolicy`.

Reply:
I feel that this eventually requires an integration test somewhere.

Reply:
I've added a test to the discovery integration suite

Reply:
Thanks @andrew-coleman for adding the integration test. 

One more point - I am not sure what was the motivation but it appears that we also allow the implicit collection policy to be [configured via channel config](https://github.com/hyperledger/fabric/blob/240cf0e182938a09832ba4162c3c229c81848e3f/core/chaincode/lifecycle/deployedcc_infoprovider.go#L266).

I feel that moving the existing code in a separate function and invoking that function from both the existing code and your code may allow to keep this important logic at one place.

Reply:
Merging this as is for now. Opened a new [issue](https://github.com/hyperledger/fabric/issues/2798) so it can be tracked.